### PR TITLE
docs: market params design_doc

### PR DESCRIPTION
**Pull request by toteki**
New ADR (as folder is not yet moved/renamed) absorbs some previous changes and proposes three similarly themed parameters:
- `MaxSupplyUtilization`
- `MaxCollateralShare`
- `MinCollateralLiquidity` (alternative: `MaxCollateralUtilization`)

This has been split off from #1141 into a docs-only PR for process.

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] added `!` to the type prefix if API or client breaking change
- [x] added appropriate labels to the PR
- [x] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] provided a link to the relevant issue or specification
- [x] added a changelog entry to `CHANGELOG.md`
- [x] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [x] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: docs/architecture/ADR-003-borrow-assets.md
Comment (robert-zaremba) [inline docs/architecture/ADR-003-borrow-assets.md:50]: Why moving this definitions?

File: docs/architecture/ADR-010-market-params.md
Comment (robert-zaremba) [inline docs/architecture/ADR-010-market-params.md:1]: Is this going to specify a task or be a documentation? If latter, then we should organize it as we discussed recently. If former, then we should not re-copy definitions, and only include here what's new. Also, probably it should be formulated as the design doc.

File: docs/architecture/ADR-010-market-params.md
Comment (robert-zaremba) [inline docs/architecture/ADR-010-market-params.md:98]: so the definition is the same? Nothing changed.

**Review by robert-zaremba — COMMENTED** [review]
Is there anything new in this PR? 
* I thought you were mentioning to reverse `collateral_utilization`. It stays the same in this PR. Also, as noted in https://github.com/umee-network/umee/pull/940, reversing doesn't help. I think you are noting this in this PR as well (in alternatives section). Commenting about it because I thought that this was the motivation to block https://github.com/umee-network/umee/issues/1036.
* If this is going to improve documentation, then let's put it as a documentation (in x/leverage/spec)

File: docs/architecture/ADR-003-borrow-assets.md
Comment (toteki) [reply docs/architecture/ADR-003-borrow-assets.md:50]: It was not part of the original decision - since ADR-003 will be renamed to a design doc (and already functioned as one), this is an attempt to apply the rule "don't modify old design docs, but make a new one instead."

Since multiple parameters are being added in the same timeframe, it made sense to group the proposals.

File: docs/architecture/ADR-010-market-params.md
Comment (toteki) [reply docs/architecture/ADR-010-market-params.md:1]: This is a design doc - it outlines decision + motivation + alternatives.

File: docs/architecture/ADR-010-market-params.md
Comment (toteki) [reply docs/architecture/ADR-010-market-params.md:98]: This is proposing using `MinCollateralLiquidity = 0.2` instead of `MaxCollateralUtilization = 5.0`. (Hence collateral utilization appearing in the `# Alternatives` section.

Functionally the same, but `MinCollateralLiquidity` will range 0-1 (helping with basis points), and the naming is different.

It does not aim to address the `1/N` curve for `utilization -> interest rate` as that is unchanged, and out of scope for this design doc.

**Comment by toteki** [comment]
> Is there anything new in this PR?

- Adds two new parameters `MaxCollateralShare` and `MaxSupplyUtilization`
- Renames + redefines `MaxCollateralUtilization` to `MinCollateralLiquidity`
- Moves the latter to a new doc with the others

**Comment by toteki** [comment]
This is the first PR that was split off from #1141, and should ideally be merged before #1159.

File: docs/architecture/ADR-003-borrow-assets.md
Comment (robert-zaremba) [reply docs/architecture/ADR-003-borrow-assets.md:50]: I know, but they are related to the borrowing as well.
Hence, if we are moving this then let's already organize it in a documentation, rather than scatter it around decision docs.

**Comment by robert-zaremba** [comment]
1. It would be great if we already start moving towards a nice documentation flow, and link definitions and whatever is needed into the design docs. If we desided that the design docs are going to be static (won't change after the design is settled), then we end up with scattered pieces distributed among decision discussed in desig docs.
2. For completeness, let's already place this document in `docs/design_docs`

File: docs/architecture/ADR-003-borrow-assets.md
Comment (toteki) [reply docs/architecture/ADR-003-borrow-assets.md:50]: What specifically should I do to this PR before merging? As it stands, it is a completed `design doc` representing parameters which were added together on a single theme.

Moving the collateral utilization definition back to 003 violates "don't modify old design docs" rule. Removing this document altogether and putting it in the spec goes against our decision to use design doc PRs.

As I see it, this should either be merged unchanged, or moved to the design docs folder first then merged. Spec updates are a separate issue and do not happen during the initial design PR.

File: docs/design_docs/010-market-params.md
Comment (adamewozniak) [inline docs/design_docs/010-market-params.md:6]:
```suggestion
- Jul 18, 2022: Initial draft moved discussion to ADR-010 (@toteki)
```

File: docs/design_docs/010-market-params.md
Comment (adamewozniak) [inline docs/design_docs/010-market-params.md:98]:
```suggestion
collateral_liquidity(token) = available(token) / total_collateral(token) // [0, ∞]
```

File: docs/design_docs/010-market-params.md
Comment (adamewozniak) [inline docs/design_docs/010-market-params.md:128]:
```suggestion
collateral_utilization(token) = total_collateral(token) / available(token) // [0 - ∞]
```

File: docs/design_docs/010-market-params.md
Comment (adamewozniak) [inline docs/design_docs/010-market-params.md:135]:
```suggestion
Stable assets will have high max collateral utilization (up to to 50). Volatile assets should have significantly smaller collateral utilization, and assets with high risk should have max collateral utilization close to 1.
```

**Review by adamewozniak — APPROVED** [review]

**Comment by robert-zaremba** [comment]
I hoped we won't merge PRs without updated documentation (`spec` dir)

**Comment by toteki** [comment]
> I hoped we won't merge PRs without updated documentation (`spec` dir)

Current practice is to update spec simultaneous with implementation, so it is always exactly up to date (as opposed to containing unimplemented features).

As an example, the `proto` update with these parameters also added them to the `Token` struct in the spec.

We will likely update the descriptions (specifically failure conditions) of `Msg` types in the spec, in the same PR as those limits are introduced in code.

Also since spec is scheduled for overhaul (to be fleshed out to be an adequate source of truth) we can wait until v3 release candidate is out to do the large spec updates.

If the above turns out to be bad practice, we can revisit it - for now, I want to rush v3.

File: docs/design_docs/010-market-params.md
Comment (toteki) [reply docs/design_docs/010-market-params.md:6]: Note: Fixed in document moving/renaming PR

**Comment by robert-zaremba** [comment]
> Current practice is to update spec simultaneous with implementation, so it is always exactly up to date (as opposed to containing unimplemented features).

I have never seen that. The whole motivation to redo the adr process and put more pressure to the documentation was that both were and still are very outdated.

> We will likely update the descriptions (specifically failure conditions) of Msg types in the spec, 

We should really link it to the code / or proto file rather than copying it. This way we are sure the we will be up to date.

**Comment by robert-zaremba** [comment]
True - we don't need to update the docs now, we don't have that task. But if in any current tasks we are updating something then we should already slowly construct the updated docs.

**Pull request merged**
